proto/internal/temporal/server/api/persistence/v1/executions.proto

```diff
@@ -294,6 +294,20 @@ message WorkflowExecutionState {
     temporal.api.enums.v1.WorkflowExecutionStatus status = 4;
     VersionedTransition last_update_versioned_transition = 5;
     google.protobuf.Timestamp start_time = 6;
+    // New fields added to WorkflowExecutionState should be added to WorkflowExecutionStateDetails
+    // to avoid modifying the SQL table schema.
+    WorkflowExecutionStateDetails details = 7;
```

Comment:
not a huge fan of of this... I feel this should be handled by the sql execution store impl (create this separate details message with certain fields in ExecutionState definition and do an extra encoding step.) instead of moving the concern up the layer.

I guess I still hope eventually, the current execution table will just have one blob column which has everything here. With the current approach, I don't feel that will happen as we are pushing all new fields to the Details message and end up with a weird definition for WorkflowExecutionState (only one Details field with everything in it.)

Reply:
Changed so it's just a SQL implementation detail.